icu4c/source/common/cjmlbe.cpp

```diff
@@ -0,0 +1,462 @@
+    }
+    if (w4 != INVALID && w5 != INVALID && w6 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        arr[2] = w6;
+        arr[3] = -1;
+        featureList->adoptElement(concatChar("TW4:", arr, status), status);
+    }
+    if (b1 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB1:", b1, status), status);
+    }
+    if (b2 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB2:", b2, status), status);
+    }
+    if (b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB3:", b3, status), status);
+    }
+    if (b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB4:", b4, status), status);
+    }
+    if (b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB5:", b5, status), status);
+    }
+    if (b6 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB6:", b6, status), status);
+    }
+    if (b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("BB1:", UnicodeString(b2).append(b3), status), status);
+    }
+    if (b3 != UnicodeString(INVALID) && b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("BB2:", UnicodeString(b3).append(b4), status), status);
+    }
+    if (b4 != UnicodeString(INVALID) && b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("BB3:", UnicodeString(b4).append(b5), status), status);
+    }
+    if (b1 != UnicodeString(INVALID) && b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB1:", UnicodeString(b1).append(b2).append(b3), status), status);
+    }
+    if (b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID) && b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB2:", UnicodeString(b2).append(b3).append(b4), status), status);
+    }
+    if (b3 != UnicodeString(INVALID) && b4 != UnicodeString(INVALID) && b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB3:", UnicodeString(b3).append(b4).append(b5), status), status);
+    }
+    if (b4 != UnicodeString(INVALID) && b5 != UnicodeString(INVALID) && b6 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB4:", UnicodeString(b4).append(b5).append(b6), status), status);
+    }
+
+    int32_t score = negativeSum;
+    for (int32_t j = 0; j < featureList->size(); j++) {
+        UnicodeString* key = (UnicodeString*)featureList->elementAt(j);
+        if(fModel.containsKey(*key)) {
+            score += (2 * fModel.geti(*key));
+        }
+    }
+
+    if (score > 0) {
+        boundary.addElement(index, status);
+        numBreaks++;
+    }
+
+    featureList->removeAllElements();
```

Comment:
No need to call featureList->removeAllElements()
The desctructor of featureList wil handle that.

Reply:
Done.